documentation/proposals/Proposal - 3.0 & 3.X Software Development Plan.md

```diff
@@ -0,0 +1,149 @@
+
+### 3.0 Preview 3
+
+By this preview, the groundwork has been established for 3.0 and we should ensure that all of our goals have ample progress towards the end product. In this version:
+- Bugfixes from 3.0 Preview 2
+- The Scraper is complete.
+- The Overloader is complete.
+- iOS support has been added for Windowing and Input, and have received initial testing on this platform.
+- If time permits, a start has been made on the SIMD APIs in Maths. No work has been done on integrating it into the other Maths types.
+
+3.0 Preview 3 is not a production-ready preview and is very experimental. This preview should aim to release in early October.
+
+### 3.0 Preview 4
+
+This is the first "production-ready" preview and we want users to start integrating into their workloads, so we need to make sure good progress has been made to all goals for the 3.0 update and as many forseeable breaking changes as possible done. In this version:
+- Bugfixes from 3.0 Preview 3
+- Windowing integrations for WPF and WinForms have been developed and have at least basic OpenGL support. The support may not be the most high performance possible at this time.
+- SIMD APIs in Maths have been complete, and work has started to integrate them into the other Maths types in the most common cases.
+- Ample work has been done to migrate 2.0 code to 3.0 code to evaluate differences in public API, fixing them where we deem necessary.
+
+3.0 Preview 4 is a production-ready preview and users are encouraged to start integrating this preview into their code. This preview should aim to release in late October.
+
+### 3.0 Preview 5
+
+This is the last preview and is primarily a bugfix release. All breaking changes should've been done in previous previews, but if this is not the case all forseeable breaking must be 100% done in this preview. In this version:
+- Bugfixes from 3.0 Preview 4
+- A windowing integration for MAUI has been developed and has at least basic OpenGL(ES) support in a state that is as high-performance and as smoothed-out as possible.
+- If time permits, a windowing integration for Avalonia has been developed and has at least basic OpenGL(ES) support. If there is not enough time, this can be pushed to 3.X.
+- SIMD APIs should be integrated into Maths in as many common cases as possible. Ongoing performance improvements may be done in 3.X. This preview should aim to release in early November.
+
+## Problems identified in past development
+
+- We have severely lacking documentation
+    - The intention is that all developers of large amounts of code write implementation documentation and/or "orientation guides" for their codebases informing readers of all major things there is to know in their code.
+    - We should also write documentation containing examples on using as many features of the surface APIs as possible \[for high level utilities\]
```

Comment:
Can we do this inline?

Reply:
Mit relevant to this proposal, but maybe you can explain how this works - I trust you are right, but I don't know how we'll do it.

Reply:
Sure, documented more.